### PR TITLE
add mixed optimization for list optimization

### PR DESCRIPTION
**Pull request by jduerholt**
## Motivation

As outlined in issue https://github.com/pytorch/botorch/issues/1272, `optimize_acqf_list` cannot be used to optimize over mixed domains. For this reason, this PR introduces the argument `fixed_features_list` for `òptimize_acqf_list`. Calling `optimize_acqf_list` with a list of fixed features invokes `optimize_acqf_mixed` instead of `optimize_acqf` under the hood. 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/pytorch/botorch/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

In principle unit tests, but I had problems to understand how the tests of the different optimisation functions work, perhaps @Balandat or @saitcakmak can help me here ;)

### CODE REVIEW

**Review by saitcakmak — COMMENTED** [review]
Thanks for putting this together and sorry for the late review!

Re testing: You can mock out `optimize_acqf` and `optimize_acqf_mixed`, and check that the correct one was called with the correct arguments. [Here's](https://github.com/pytorch/botorch/blob/main/test/optim/test_optimize.py#L93-L115) an example from tests that mock `gen_candidates_scipy` and `gen_batch_initial_conditions`.  You can then check `call_args / call_args_list` of the mock object to assert that it got called with the correct arguments. Lmk if you need more specific examples.

File: botorch/optim/optimize.py [hunk 1]
```diff
@@ -409,6 +410,10 @@ def optimize_acqf_list(
             `\sum_i (X[indices[i]] * coefficients[i]) = rhs`
         fixed_features: A map `{feature_index: value}` for features that
             should be fixed to a particular value during generation.
+        fixed_features_list: A list of maps `{feature_index: value}`. The i-th
+            item represents the fixed_feature for the i-th optimization. If
+            `fixed_features_list` is provided, `optimize_acqf_mixed` is invoked
+            and the `fixed_features` argument is ignored.
```
[the remark this answers is not in the document]
Comment (jduerholt) [reply botorch/optim/optimize.py]: I changed the behavior according to your suggestion.

**Comment by jduerholt** [comment]
@saitcakmak: I implemented the test according to how I understood your suggestion. Can you have a look?

**Review by saitcakmak — APPROVED** [review]
Thanks, lgtm!

**Pull request closed without merging**